### PR TITLE
GEODE-6883: New module for Membership

### PR DESCRIPTION
**Pull request by echobravopapa**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/CacheOperationMessageMarker.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.distributed.internal.membership.api;
+
+/**
+ * This interface in implemented by message classes performing cache
+ * operations and is used to check for message type internally.
+ */
+public interface CacheOperationMessageMarker {
```
Comment (upthewaterspout) [inline geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/CacheOperationMessageMarker.java]: I think we should pull this checking out of membership. Even if we leave this marker in, the javadocs above don't really describe what this is doing or why someone would add this marker to a message.

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java [hunk 2]
```diff
@@ -582,7 +581,7 @@ boolean doTCPCheckMember(ID suspectMember, int port,
       } catch (IOException e) {
         // this is expected if it is a connection-timeout or other failure
         // to connect
-      } catch (IllegalStateException | GemFireConfigException e) {
+      } catch (IllegalStateException e) {
```
Comment (upthewaterspout) [inline geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java]: Should we remove this whole catch block? If it was good to propage GemFireConfigException up, maybe we should do that for IllegalStateException as well?

Comment (echobravopapa) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java]: we could do that... only action we took was to potentially log a small message...

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/CacheOperationMessageMarker.java (hunk 1 above)
Comment (echobravopapa) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/CacheOperationMessageMarker.java]: I'd be happy with pulling this out, maybe on a separate ticket...?

Comment (upthewaterspout) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/CacheOperationMessageMarker.java]: We can do this as a separate ticket if you prefer.

Comment (echobravopapa) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/CacheOperationMessageMarker.java]: https://issues.apache.org/jira/browse/GEODE-7695

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java (hunk 2 above)
Comment (echobravopapa) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java]: https://issues.apache.org/jira/browse/GEODE-7696

**Pull request merged**